src/dove/core/system.py

```diff
@@ -103,42 +106,78 @@ def summary(self) -> None:
         }
         print(info)
 
-    def verify(self) -> None:
+    def verify_components_definition(self) -> None:
         """
-        Verify the integrity of the system.
+        Verify that components are valid.
 
         Checks for:
+        - Type of components
         - Uniqueness of component names
-        - Uniqueness of resource names
-        - Consistency of time series lengths with the system time index
 
         Raises
         ------
+        TypeError
+            If any component is not of type Component
         ValueError
-            If any validation check fails:
-            - Component names are not unique
-            - Resource names are not unique
-            - Component profile length doesn't match time index length
-            - Cashflow price profile length doesn't match time index length
+            If component names are not unique
         """
+        # Check the types of the components
+        for comp in self.components:
```

Comment:
I don't think we need to explicitly check that it's a component. If we request `comp.name` it will throw an error if it's not a component. This is locking down the development of the system too. much.